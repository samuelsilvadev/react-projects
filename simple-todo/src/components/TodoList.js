import React from 'react';
import './css/Todo.css';
import TodoItem from './TodoItem'

const TodoList = ({ todos }) => {

    const nodeTodos = todos.map((todo) => {
        return <TodoItem key={todo.id} value={todo.todo} />
    });
    return (<ol className="todo-list" >{nodeTodos}</ol>);
}

export default TodoList;