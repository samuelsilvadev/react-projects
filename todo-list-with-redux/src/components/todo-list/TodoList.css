.todo-item {
	cursor: pointer;
}

.completed-todo {
	text-decoration: line-through;
}
