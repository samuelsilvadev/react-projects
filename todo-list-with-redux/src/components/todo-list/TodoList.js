import React from 'react';
import { connect } from 'react-redux';

import { toggleTodo } from './../../state/actions-creators';
import { SHOW_ACTIVE, SHOW_ALL, SHOW_COMPLETED } from './../../state/types';

import './TodoList.css';

const getFilteredTodos = (todos, activeFilter) => {
	return {
		[SHOW_ALL]: todos,
		[SHOW_ACTIVE]: todos.filter((todo) => !todo.completed),
		[SHOW_COMPLETED]: todos.filter((todo) => todo.completed),
	}[activeFilter] || [];
};

export const TodoList = ({ todos, toggleTodo, activeFilter }) => {

	if (todos.length === 0) {
		return null;
	}

	const _handleToggleTodo = (id) => () => {
		toggleTodo(id);
	};

	const _renderTodoItem = (todo) => {
		const { id, text, completed: isCompleted } = todo;

		const classNames = isCompleted ? 'completed-todo' : '';

		return (
			<li
				data-enzyme-id={ `todo-${id}` }
				className={ `collection-item todo-item ${classNames}` } key={ id } onClick={ _handleToggleTodo(id) }>
				{ text }
			</li>
		);
	};

	return (
		<ul
			data-enzyme-id="todo-wrapper"
			className="collection">
			{ getFilteredTodos(todos, activeFilter).map(_renderTodoItem) }
		</ul>
	);
}

TodoList.defaultProps = {
	todos: [],
	activeFilter: SHOW_ALL,
};

const mapStateToProps = (state) => ({
	todos: state.todos,
	activeFilter: state.visibilityFilters,
});

const mapDispatchToProps = (dispatch) => ({
	toggleTodo: (id) =>  dispatch(toggleTodo(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TodoList);
