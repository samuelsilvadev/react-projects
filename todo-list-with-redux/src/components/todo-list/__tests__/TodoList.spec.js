import React from 'react';
import { shallow } from 'enzyme';

import { TodoList } from './../TodoList';

const todos = [{
	id: 1,
	text: 'Go to supermarket',
	completed: false,
}];

describe('<TodoList />', () => {
	it('should render correctly', () => {
		
		const tree = shallow(<TodoList todos={ todos } />);
		
		expect(tree.exists('[data-enzyme-id="todo-wrapper"]')).toBe(true);
		expect(tree.exists('[data-enzyme-id="todo-1"]')).toBe(true);
		expect(tree).toMatchSnapshot();
	});

	it('should render nothing without todos', () => {
		const tree = shallow(<TodoList />);
		
		expect(tree.exists('[data-enzyme-id="todo-wrapper"]')).toBe(false);
	});

	it('should call `toggleTodo` when click on todo item', () => {
		const toggleTodo = jest.fn();
		const tree = shallow(<TodoList todos={ todos } toggleTodo={ toggleTodo } />);
		const $todo = tree.find('[data-enzyme-id="todo-1"]');

		$todo.simulate('click');
		
		expect(toggleTodo).toHaveBeenCalledTimes(1);
		expect(toggleTodo).toHaveBeenCalledWith(1);
	});
});
