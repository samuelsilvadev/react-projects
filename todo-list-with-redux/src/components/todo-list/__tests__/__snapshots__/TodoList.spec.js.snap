// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    activeFilter="visibilityFilter:SHOW_ALL"
    todos={
      Array [
        Object {
          "completed": false,
          "id": 1,
          "text": "Go to supermarket",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="collection-item todo-item "
          data-enzyme-id="todo-1"
          onClick={[Function]}
        >
          Go to supermarket
        </li>,
      ],
      "className": "collection",
      "data-enzyme-id": "todo-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Go to supermarket",
          "className": "collection-item todo-item ",
          "data-enzyme-id": "todo-1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Go to supermarket",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="collection-item todo-item "
            data-enzyme-id="todo-1"
            onClick={[Function]}
          >
            Go to supermarket
          </li>,
        ],
        "className": "collection",
        "data-enzyme-id": "todo-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Go to supermarket",
            "className": "collection-item todo-item ",
            "data-enzyme-id": "todo-1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Go to supermarket",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
